torch/distributed/fsdp/flat_param.py

```diff
@@ -1356,41 +1484,63 @@ def _use_sharded_flat_param(self) -> None:
     #########
     # VIEWS #
     #########
-    @staticmethod
-    def _get_unflat_views(
-        flat_param: FlatParameter,
+    @no_type_check
+    def _get_unflat_views_unaligned(
+        self,
         tensor: Optional[torch.Tensor] = None,
     ) -> Iterator[Tensor]:
         """
         Returns unflattened ``Tensor`` views into ``tensor`` if it is not
         ``None`` or ``flat_param`` otherwise, where the unflattening is based
         on ``flat_param`` 's metadata.
 
-        In other words, to get views into the unsharded flat parameter, pass
-        ``tensor`` as ``None``, but to get views into tensor optimizer state,
-        pass ``tensor`` as the optimizer state tensor.
+        Examples for ``tensor`` include ``flat_param.grad`` or unsharded
+        tensor optimizer state.
         """
+        flat_param = self.flat_param
         if tensor is None:
             tensor = flat_param
-        _p_assert(
-            tensor.numel() == flat_param._unpadded_unsharded_size.numel(),
-            f"Expects {flat_param._unpadded_unsharded_size.numel()} numel but got "
-            f"{tensor.numel()} numel",
-        )
         views = (
             _ext_post_unflatten_transform(subtensor.view(shape), param_extension)
             for (subtensor, shape, param_extension) in zip(
-                torch.split(tensor, flat_param._numels, dim=0),  # type: ignore[arg-type]
+                torch.split(tensor, flat_param._numels, dim=0),
                 flat_param._shapes,
                 flat_param._param_extensions,
             )
         )
         return views
 
+    @no_type_check
+    def _get_unflat_views_aligned(
+        self,
+        tensor: Optional[Tensor] = None,
+    ) -> List[Optional[Tensor]]:
+        """
+        This has the same contract as :meth:`_get_unflat_views_unaligned`
+        except it checks for ``None`` placeholders representing padding for
+        alignment, which may incur slightly more CPU overhead.
```

Comment:
nit: add docs about the input args?

Reply:
I am wondering, should we repeat the comment from `_get_unflat_views_unaligned()` or just refer developers to it?

I worry about maintaining two sets of comments.